Geometry_Engine/Modify/SplitAtPoints.cs

```diff
@@ -212,111 +212,113 @@ public static List<PolyCurve> SplitAtPoints(this PolyCurve curve, List<Point> po
                 return new List<PolyCurve> { curve.DeepClone() };
 
             List<PolyCurve> result = new List<PolyCurve>();
-            List<ICurve> tmpResult = new List<ICurve>();
-            List<Point> subPoints = new List<Point>();
-            List<Point> onCurvePoints = new List<Point>();
 
-            foreach (Point p in points)
+            List<ICurve> subParts = curve.ISubParts().ToList(); //Subparts of PolyCurve
+            if (subParts.Count == 1)    //Single SubPart
```

Comment:
What's the benefit of having a dedicated case for a single segment? Couldn't it be processed by the general case?

Reply:
Can probably refactor it slightly to work, but think there are some cases that need to be handled slightly differently, for example, if it is split at start point, then the general case simply adds the segment to the list, while the special case for single segment instead adds the segment to a new "prev".

Slight difference in handling of end point split as well.

Basically, general cases handles 
1. Start split 
2. all middle splits
3. End split 

all separately

While the case for a single split it should only handle start _or_ end split case, if that makes sense?

Reply:
It does, but middle split is handled starting from `s = 1`, which will never be the case for a single split, so assuming that start/end should not be dependent on the number of resultant segments, it could be generalised?

Reply:
Had another look, and managed to refactor it all into something with less edge cases. THink this made the code cleaner (and significantly shorter) so thanks for pushing @pawelbaran .